src/vmm/src/devices/virtio/device.rs

```diff
@@ -54,14 +54,14 @@ pub enum IrqType {
 /// Helper struct that is responsible for triggering guest IRQs
 #[derive(Debug)]
 pub struct IrqTrigger {
-    pub(crate) irq_status: Arc<AtomicUsize>,
+    pub(crate) irq_status: Arc<AtomicU32>,
```

Comment:
According to the virtio spec (Section 4.2.2), this value is either 0 or 1, so I'm not sure why this was defined as `AtomicUsize` (a quick glance at the git history didnt tell me much either). I think this change is fine, and the require snapshot version bump here is not a problem. 

@bchalios do you have any idea what might've been the reason for making this a usize?

Reply:
Indeed, it's either 0 or 1 but what's its length? I couldn't find that in the spec.

Reply:
>

Interrupt status
Reading from this register returns a bit mask of events that caused the device interrupt to be asserted. The following events are possible:
- Used Buffer Notification: bit 0 - the interrupt was asserted because the device has used a buffer in at least one of the active virtual queues.
- Configuration Change Notification: bit 1 - the interrupt was asserted because the configuration of the device has changed.

Reply:
I think its implicit by virtue of being mapped to a single MMIO register, and each MMIO register is 32bits wide (e.g. the interrupt_status is at 0x60, and at 0x64 a different field starts, page 59 of the 1.1 spec, Table 4.2)

Reply:
https://docs.oasis-open.org/virtio/virtio/v1.1/csprd01/virtio-v1.1-csprd01.html#x1-1460002
search for 
```
InterruptStatus
0x60
R
```

Reply:
Ah, you're right, I misread. Its not always 0 or 1, but instead only the bits 0 and 1 have meaning. So its a 2 bit field, thanks Nikita! :D

Reply:
I did some research. Qemu stores this as a `u8`: https://github.com/qemu/qemu/blob/005ad32358f12fe9313a4a01918a55e60d4f39e5/include/hw/virtio/virtio.h#L109

In any case, we write back to the guest a 32bit value: https://github.com/firecracker-microvm/firecracker/blob/e2a9ba42d8fe72590d9ca8858be3d40cfcf7e3ab/src/vmm/src/devices/virtio/mmio.rs#L249, so we should be fine with `AtomicU32`